src/util/mod.rs

```diff
@@ -149,22 +149,45 @@ pub fn to_socket_addr<A: ToSocketAddrs>(addr: A) -> io::Result<SocketAddr> {
 /// A struct to make byte array format more friendly.
 ///
 /// Remove this once issue [33127](https://github.com/rust-lang/rust/issues/33127) is resolved.
-pub struct HexDisplay<'a> {
+pub struct PrettyDispaly<'a> {
     data: &'a [u8],
 }
 
-impl<'a> Display for HexDisplay<'a> {
+
+
+impl<'a> Display for PrettyDispaly<'a> {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
-        try!(write!(f, "0x"));
-        for &b in self.data {
-            try!(write!(f, "{:02X}", b));
-        }
+        let mut buf = String::new();
+        print_bytes_to(self.data, &mut buf);
+        write!(f, "{}", buf).unwrap();
         Ok(())
     }
 }
 
-pub fn hex(data: &[u8]) -> HexDisplay {
-    HexDisplay { data: data }
+/// for printing the same format with protobuf
+/// original implementation 
+/// https://github.com/stepancheg/rust-protobuf/blob/8e19c644c7d744c79d7bc097a0861baccecf586b/src/lib/text_format.rs#L8
+fn print_bytes_to(bytes: &[u8], buf: &mut String) {
```

Comment:
Why not move to `PrettyDisplay::display`? So we can use `write!` rather than creating a new string.